tests/dp_engine_test.py

```diff
@@ -137,6 +153,72 @@ def test_aggregate_report(self):
         engine.aggregate(None, params2, None)
         self.assertEqual(len(engine._report_generators), 2)  # pylint: disable=protected-access
 
+    def _mock_and_assert_private_partitions(self, engine: pipeline_dp.DPEngine, groups, 
+                                            min_users, expected_partitions,
+                                            max_partitions_contributed):
+        with patch("pipeline_dp.dp_engine.create_truncated_geometric_partition_strategy",
+                   new=_mock_partition_strategy_factory(min_users)) as mock_factory:
+            data_filtered = engine._select_private_partitions(groups, max_partitions_contributed)
+            engine._budget_accountant.compute_budgets()
+            self.assertListEqual(list(data_filtered), expected_partitions)
+
+    def test_select_private_partitions(self):
+        input_col = [
+            ("pid1", ( 'pk1', 1 )), 
```

Comment:
You can create MockAccumulator which has private_id_count and create input_col of (partition_key, MockAccumulator). MockAccumulator might be NamedTuple for example.

And then no need to do group_by_key below

Reply:
Done something equivalent.

Reply:
ok, it makes sense.